src/nvim/window.c

```diff
@@ -5421,6 +5430,29 @@ static win_T *restore_snapshot_rec(frame_T *sn, frame_T *fr)
   return wp;
 }
 
+win_T *get_snapshot_focus_rec(frame_T *sn)
+{
+  /* This should be equivalent to the recursive algorithm found in
+   * restore_snapshot as far as traveling nodes go. */
+  while (sn->fr_child != NULL || sn->fr_next != NULL) {
+    while (sn->fr_child != NULL) {
+      sn = sn->fr_child;
+    }
+    if (sn->fr_next != NULL) {
+      sn = sn->fr_next;
+    }
+  }
+
+  return sn->fr_win;
+}
+
+/* Get the focused window (the one holding the cursor) of the snapshot */
+win_T *get_snapshot_focus(int idx)
+{
+  if (curtab->tp_snapshot[idx] == NULL)
+    return NULL;
+  return get_snapshot_focus_rec(curtab->tp_snapshot[idx]);
```

Comment:
Since it's not recursive, there's no need for a separate function (and the `_rec` suffix).